nilearn/reporting/glm_reporter.py

```diff
@@ -967,3 +972,221 @@ def _dataframe_to_html(df, precision, **kwargs):
+        String to prepend to generated filenames.
+        If a string is provided, "_" will be added to the end.
+        Default is None.
+
+    Notes
+    -----
+    To output:
+    - design matrix
+    - contrast plot
+    - contrast-level stat maps
+    - model-level stat maps
+    - metadata
+    """
+    if isinstance(prefix, str) and not prefix.endswith('_'):
+        prefix += '_'
+    else:
+        prefix = ''
+
+    out_dir = os.path.abspath(out_dir)
+
+    # Write out design matrices to files.
+    if hasattr(model, 'design_matrices_'):
+        design_matrices = model.design_matrices_
+    else:
+        design_matrices = [model.design_matrix_]
+
+    # TODO: Assuming that cases of multiple design matrices correspond to
+    # different runs. Not sure if this is correct. Need to check.
+    for i_run, design_matrix in enumerate(design_matrices):
+        run_str = (
+            "run-{}_".format(i_run + 1) if len(design_matrices) > 1 else ""
+        )
+        dm_file = os.path.join(
+            out_dir,
+            '{}{}design.tsv'.format(prefix, run_str),
+        )
+        design_matrix.to_csv(
+            dm_file,
+            sep='\t',
+            line_terminator='\n',
+            index=False,
+        )
+
+        dm_fig_file = os.path.join(
+            out_dir,
+            '{}{}design.svg'.format(prefix, run_str),
+        )
+        dm_fig = plot_design_matrix(design_matrix)
+        dm_fig.figure.savefig(dm_fig_file)
+
+        # Save contrast plots as well
+        for contrast_name, contrast_data in contrasts.items():
+            contrast_plot = plot_contrast_matrix(
+                contrast_data,
+                design_matrix,
+                colorbar=True,
+            )
+            contrast_plot.set_xlabel(contrast_name)
+            contrast_plot.figure.set_figheight(2)
+            contrast_plot.figure.set_tight_layout(True)
+            contrast_name = _clean_contrast_name(contrast_name)
+            constrast_fig_file = os.path.join(
+                out_dir,
+                "{}{}contrast-{}_design.svg".format(
+                    prefix,
+                    run_str,
+                    contrast_name,
+                )
+            )
+            contrast_plot.figure.savefig(constrast_fig_file)
+
+    statistical_maps = _make_stat_maps(
+        model,
+        contrasts,
+        output_type='all',
+    )
+
+    # Model metadata
+    # TODO: Determine optimal mapping of model metadata to BIDS fields.
+    metadata_file = os.path.join(out_dir, 'dataset_description.json')
+
+    SELECTED_ATTRIBUTES = [
+        'subject_label',
+        'drift_model',
+        'hrf_model',
+        'standardize',
+        't_r',
+        'high_pass',
+        'target_shape',
+        'signal_scaling',
+        'drift_order',
+        'scaling_axis',
+        'smoothing_fwhm',
+        'target_affine',
+        'slice_time_ref',
+        'fir_delays',
+    ]
+    # ATTRIBUTE_UNITS = {
```

Comment:
to be removed ?

Reply:
I think that the sidecar JSON files (not yet written out) should include unit information, but the exact metadata to retain,  how the fields should be named, etc. is not yet determined.

Reply:
In retrospect, it's probably not worth the effort to try to incorporate units just yet, so I'll remove.